File: validate_proxy_plugin.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/mattn/go-sqlite3"
)

func main() {
	// 连接到数据库
	db, err := sql.Open("sqlite3", "./data/data.db")
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()

	fmt.Println("=== 代理插件完整功能验证 ===")

	// 1. 检查表结构
	fmt.Println("\n1. 检查traders表结构:")
	rows, err := db.Query("PRAGMA table_info(traders)")
	if err != nil {
		log.Fatal(err)
	}
	defer rows.Close()

	hasDataAccessMethod := false
	for rows.Next() {
		var cid int
		var name, typ string
		var notnull, pk int
		var dflt_value sql.NullString
		err := rows.Scan(&cid, &name, &typ, &notnull, &dflt_value, &pk)
		if err != nil {
			log.Fatal(err)
		}
		fmt.Printf("   %d. %s (%s)\n", cid, name, typ)
		if name == "data_access_method" {
			hasDataAccessMethod = true
			defaultVal := ""
			if dflt_value.Valid {
				defaultVal = dflt_value.String
			}
			fmt.Printf("      ✅ 代理设置字段存在，默认值: '%s'\n", defaultVal)
		}
	}

	if !hasDataAccessMethod {
		fmt.Println("   ❌ 未找到data_access_method字段!")
		return
	}

	// 2. 检查现有数据
	fmt.Println("\n2. 检查现有交易员代理设置:")
	traderRows, err := db.Query(`
		SELECT id, name, data_access_method, exchange_id, ai_model_id 
		FROM traders 
		ORDER BY created_at DESC
	`)
	if err != nil {
		log.Fatal(err)
	}
	defer traderRows.Close()

	count := 0
	proxyCount := 0
	nativeCount := 0

	for traderRows.Next() {
		var id, name, dataAccessMethod, exchangeID, aiModelID string
		err := traderRows.Scan(&id, &name, &dataAccessMethod, &exchangeID, &aiModelID)
		if err != nil {
			log.Fatal(err)
		}

		count++
		fmt.Printf("\n   交易员 %d:\n", count)
		fmt.Printf("     名称: %s\n", name)
		fmt.Printf("     ID: %s\n", id)
		fmt.Printf("     代理设置: %s\n", dataAccessMethod)
		fmt.Printf("     交易所: %s\n", exchangeID)
		fmt.Printf("     AI模型: %s\n", aiModelID)

		if dataAccessMethod == "proxy" {
			fmt.Printf("     🟢 使用代理模式\n")
			proxyCount++
		} else {
			fmt.Printf("     🔴 使用直连模式\n")
			nativeCount++
		}
	}

	fmt.Printf("\n=== 统计结果 ===\n")
	fmt.Printf("总交易员数: %d\n", count)
	fmt.Printf("使用代理: %d\n", proxyCount)
	fmt.Printf("使用直连: %d\n", nativeCount)

	// 3. 功能验证建议
	fmt.Println("\n=== 插件功能验证建议 ===")
	fmt.Println("1. 前端验证:")
	fmt.Println("   - 访问 http://localhost:3000/traders")
	fmt.Println("   - 编辑任一交易员，检查'数据获取方式'选项")
	fmt.Println("   - 选择'代理获取'并保存")

	fmt.Println("\n2. 后端验证:")
	fmt.Println("   - 重启后端服务")
	fmt.Println("   - 查看日志中'[proxy]'相关输出")
	fmt.Println("   - 测试AI半自动工作流获取提示词")

	fmt.Println("\n3. 代理服务验证:")
	fmt.Println("   - 确保代理服务运行在 http://localhost:8081")
	fmt.Println("   - 测试代理服务是否能正常转发请求")

	fmt.Println("\n✅ 代理插件架构符合最小侵入式设计原则!")
	fmt.Println("   - 仅在两处关键位置添加代理判断")
	fmt.Println("   - 不修改原有交易所API调用逻辑")
	fmt.Println("   - 通过ProxyTraderWrapper统一处理代理转发")
}

File: validate_proxy_plugin.py
import sqlite3

def validate_proxy_plugin():
    print("=== 代理插件完整功能验证 ===")
    
    # 连接数据库
    db_path = './data/data.db'
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # 1. 检查表结构
    print("\n1. 检查traders表结构:")
    cursor.execute("PRAGMA table_info(traders)")
    columns = cursor.fetchall()
    
    has_data_access_method = False
    for col in columns:
        cid, name, typ, notnull, dflt_value, pk = col
        print(f"   {cid}. {name} ({typ})")
        if name == "data_access_method":
            has_data_access_method = True
            print(f"      ✅ 代理设置字段存在，默认值: '{dflt_value}'")
    
    if not has_data_access_method:
        print("   ❌ 未找到data_access_method字段!")
        return
    
    # 2. 检查现有数据
    print("\n2. 检查现有交易员代理设置:")
    cursor.execute("""
        SELECT id, name, data_access_method, exchange_id, ai_model_id 
        FROM traders 
        ORDER BY created_at DESC
    """)
    
    traders = cursor.fetchall()
    count = len(traders)
    proxy_count = 0
    native_count = 0
    
    for i, trader in enumerate(traders, 1):
        trader_id, name, data_access_method, exchange_id, ai_model_id = trader
        print(f"\n   交易员 {i}:")
        print(f"     名称: {name}")
        print(f"     ID: {trader_id}")
        print(f"     代理设置: {data_access_method}")
        print(f"     交易所: {exchange_id}")
        print(f"     AI模型: {ai_model_id}")
        
        if data_access_method == "proxy":
            print("     🟢 使用代理模式")
            proxy_count += 1
        else:
            print("     🔴 使用直连模式")
            native_count += 1
    
    print(f"\n=== 统计结果 ===")
    print(f"总交易员数: {count}")
    print(f"使用代理: {proxy_count}")
    print(f"使用直连: {native_count}")
    
    # 3. 功能验证建议
    print("\n=== 插件功能验证建议 ===")
    print("1. 前端验证:")
    print("   - 访问 http://localhost:3000/traders")
    print("   - 编辑任一交易员，检查'数据获取方式'选项")
    print("   - 选择'代理获取'并保存")
    
    print("\n2. 后端验证:")
    print("   - 重启后端服务")
    print("   - 查看日志中'[proxy]'相关输出")
    print("   - 测试AI半自动工作流获取提示词")
    
    print("\n3. 代理服务验证:")
    print("   - 确保代理服务运行在 http://localhost:8081")
    print("   - 测试代理服务是否能正常转发请求")
    
    print("\n✅ 代理插件架构符合最小侵入式设计原则!")
    print("   - 仅在两处关键位置添加代理判断")
    print("   - 不修改原有交易所API调用逻辑")
    print("   - 通过ProxyTraderWrapper统一处理代理转发")
    
    conn.close()

if __name__ == "__main__":
    validate_proxy_plugin()